src/main/java/seedu/address/logic/commands/EditModuleCommand.java

```diff
@@ -0,0 +1,165 @@
+import seedu.address.model.Model;
+import seedu.address.model.module.Module;
+import seedu.address.model.module.ModuleName;
+import seedu.address.model.module.student.Student;
+import seedu.address.model.task.UniqueTaskList;
+
+/**
+ * Edits the module's name
+ */
+public class EditModuleCommand extends EditCommand {
+
+    public static final String COMMAND_WORD = "edit module";
+
+    public static final String MESSAGE_NOT_EDITED = "Both old module name and new module name must be provided";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Edits a module's name. Must provide "
+            + "old module name and the new module name "
+            + "Parameters: "
+            + PREFIX_MODULE_NAME + "OLD MODULE NAME "
+            + PREFIX_NEW_MODULE_NAME + "NEW MODULE NAME "
+            + "Example: " + COMMAND_WORD + " "
+            + PREFIX_MODULE_NAME + "CS2103 "
+            + PREFIX_NEW_MODULE_NAME + "CS2105 ";
+
+    private EditModuleCommand.EditModuleDescriptor editModuleDescriptor;
+    private ModuleName moduleName;
+
+    /**
+     * Creates an EditModuleCommand to edit the specified {@code Module}
+     *
+     * @param moduleName The name of the module to edit.
+     * @param editModuleDescriptor The edited module descriptor.
+     */
+    public EditModuleCommand(ModuleName moduleName, EditModuleCommand.EditModuleDescriptor editModuleDescriptor) {
+        requireNonNull(editModuleDescriptor);
+        requireNonNull(moduleName);
+        this.editModuleDescriptor = editModuleDescriptor;
+        this.moduleName = moduleName;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        List<Module> lastShownList = model.getFilteredModuleList();
+        for (Module module : lastShownList) {
+            if (module.getName().equals(moduleName)) {
+                return editModuleInformation(model, module);
+            }
+        }
+        throw new CommandException(String.format(Messages.MESSAGE_MODULE_NAME_NOT_FOUND, moduleName.getModuleName()));
+    }
+
+    /**
+     * Edits a module's information from TAB
+     * Error checking for module not found is handled by {@code this.execute()}
+     *
+     * @param module The module to be edited
+     * @param model The TAB model that stores the list of modules
+     * @return Statement indicating that the edition is successful.
+     */
+    public CommandResult editModuleInformation(Model model, Module module) {
+        Module editedModule = createEditedModule(module, editModuleDescriptor);
+        ObservableList<Student> uniqueStudentList = module.getStudentList();
+        UniqueTaskList uniqueTaskList = module.getTaskList();
+        editedModule.setStudents(uniqueStudentList);
+        editedModule.setTasks(uniqueTaskList.asModifiableObservableList());
+        model.deleteModule(module);
+        model.addModule(editedModule);
```

Comment:
Maybe consider making a method like the Student's setStudent() method for the delete and add so it would just edit the name of the module without messing up the order of the module list?😬 because if the module is deleted and readded, it will be added to the bottom of the list. but this way is fine too I guess!

Reply:
Ahh good point! Ill change it!